// Copyright 2023 Ant Group Co., Ltd.
// 
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
// 
//   http://www.apache.org/licenses/LICENSE-2.0
// 
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

syntax = "proto3";

package envoy.extensions.filters.http.kuscia_crypt.v3;
option go_package = "gitlab.alipay-inc.com/secretflow/kuscia-envoy/kuscia/api/filters/http/crypt/v3";

message CryptRule {
    string source              = 1;
    string destination         = 2;
    string algorithm           = 3;
    string secret_key          = 4;
    string secret_key_version  = 5;
    string reserve_key         = 6;
    string reserve_key_version = 7;
}

message Crypt {
    string self_namespace            = 1;
    repeated CryptRule encrypt_rules = 2;
    repeated CryptRule decrypt_rules = 3;
}
